### PR TITLE
fix(analytics): add options for configuring pinpoint event buffer

### PR DESCRIPTION
**Pull request by 5d**
#### Description of changes

- add `EventBufferConfig` options to `PinpointProviderConfig`

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jimblanc — APPROVED** [review]
LGTM

File: packages/core/src/providers/pinpoint/utils/getEventBuffer.ts [hunk 1]
```diff
@@ -3,32 +3,40 @@
...
 	appId,
-	bufferSize = BUFFER_SIZE,
+	region,
 	credentials,
-	flushInterval = FLUSH_INTERVAL,
-	flushSize = FLUSH_SIZE,
+	bufferSize,
+	flushInterval,
+	flushSize,
+	resendLimit,
```
Comment (elorzafe) [inline packages/core/src/providers/pinpoint/utils/getEventBuffer.ts:31]: what happens with the default values?

Comment (5d) [reply packages/core/src/providers/pinpoint/utils/getEventBuffer.ts:31]: As some of the API implementations relies on the value of these options (e.g. `resentLimit` in `record` API), they've been already resolved before `getEventBuffer`. The options in `getEventBuffer` function are non-optional.

**Review by elorzafe — APPROVED** [review]
Looks good

Thanks @5d 🎖️

**Pull request merged**